lib/cloud_controller/deployment_updater/actions/scale.rb

```diff
@@ -0,0 +1,63 @@
+require 'cloud_controller/deployment_updater/actions/scale_down_superseded'
+require 'cloud_controller/deployment_updater/actions/scale_down_old_process'
+require 'cloud_controller/deployment_updater/actions/finalize'
+require 'cloud_controller/deployment_updater/calculators/all_instances_routable'
+require 'cloud_controller/deployment_updater/calculators/instances_to_scale_up'
+require 'cloud_controller/deployment_updater/calculators/instances_to_scale_down'
+
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Actions
+      class Scale
+        attr_reader :deployment, :logger, :app
+
+        def initialize(deployment, logger)
+          @deployment = deployment
+          @logger = logger
+          @app = deployment.app
+        end
+
+        def call
+          deployment.db.transaction do
+            return unless deployment.lock!.state == DeploymentModel::DEPLOYING_STATE
+            return unless Calculators::AllInstancesRoutable.new(deployment, logger).call
+
+            app.lock!
+            oldest_web_process_with_instances.lock!
+            deploying_web_process.lock!
+
+            instances_to_scale_up = Calculators::InstancesToScaleUp.new(deployment).call
+            instances_to_scale_down = Calculators::InstancesToScaleDown.new(deployment, oldest_web_process_with_instances).call
+
+            deployment.update(
+              last_healthy_at: Time.now,
+              state: DeploymentModel::DEPLOYING_STATE,
+              status_value: DeploymentModel::ACTIVE_STATUS_VALUE,
+              status_reason: DeploymentModel::DEPLOYING_STATUS_REASON
+            )
+
+            if deploying_web_process.instances >= deployment.original_web_process_instance_count
```

Comment:
**thought** Is it the case that when `instances_to_scale_up == 0` and `instances_to_scale_down == 0` we're done? I wonder if we can use the Calculator here to push some more logic in there